### PR TITLE
#2 KleisliIO code, tests and benchmarks moved from ZIO project

### LINKED ISSUE #2: Move KleisliIO from Scalaz-ZIO
Move **production code**, **tests** and **benchmarks** from `ZIO` project to this repository. 

You can find the code in:

https://github.com/scalaz/scalaz-zio/blob/master/core/shared/src/main/scala/scalaz/zio/KleisliIO.scala
https://github.com/scalaz/scalaz-zio/blob/master/core/jvm/src/test/scala/scalaz/zio/KleisliIOSpec.scala
https://github.com/scalaz/scalaz-zio/blob/master/benchmarks/src/main/scala/scalaz/zio/KleisliIOBenchmarks.scala (plus `KleisliIO` benchmark dependencies)

Include **ZIO** in `build.sbt`

```
"org.scalaz" %% "scalaz-zio" % "0.1-SNAPSHOT"
```

plus other necessary dependencies.

### PR DESCRIPTION
**Pull request by quiquedeveloper**
This PR contains all the necessary *KleisliIO* related code contained in `ZIO` project.

`build.sbt` has been configured as multi-project since benchmarks are needed

so the two main projects inside *root* are `benchmarks` and `core`

from the `sbt` command line you can run:
1. `project benchmarks`
2. `jmh:run -i 20 -wi 20 -f1 -t1 .*.`, in order to run the benchmarks
* *jmh pluging documentation can be found [here](https://github.com/ktoso/sbt-jmh)*

or

1. `project core`
2. `test`, in order to execute the tests

Fixes #2

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -21,11 +21,66 @@ lazy val sonataCredentials = for {
 
 credentials in ThisBuild ++= sonataCredentials.toSeq
 
+lazy val scalazDependencies = Seq(
+  "org.scalaz" %% "scalaz-core" % "7.2.25",
+  "org.scalaz" %% "scalaz-zio"  % "0.1-SNAPSHOT"
+)
+
```
[the remark this answers is not in the document]
Comment (quiquedeveloper) [reply build.sbt]: this does not work, i always get:
```
sbt.librarymanagement.ResolveException: unresolved dependency: org.scalaz#scalaz-zio_2.12;0.1-SNAPSHOT: not found
```

not an `sbt` expert here so maybe later we can refactor `build.sbt`

File: core/src/main/scala/scalaz/kleisliio/KleisliIO.scala [hunk 2]
```diff
@@ -0,0 +1,483 @@
+package scalaz.kleisliio
+
+import scalaz.zio.IO
+
+/**
+ * A `KleisliIO[E, A, B]` is an effectful function from `A` to `B`, which might
+ * fail with an `E`.
+ *
+ * This is the moral equivalent of `A => IO[E, B]`, and, indeed, `KleisliIO`
+ * extends this function type, and can be used in the same way.
```
Comment (leszekgruchala) [inline core/src/main/scala/scalaz/kleisliio/KleisliIO.scala]: `extends this function type` -> `applies this function type`?

File: build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (wi101) [reply build.sbt]: it should work (and remove the resolvers in settings of core & benchmarks projects)

[the remark this answers is not in the document]
Comment (wi101) [reply build.sbt]: @quiquedeveloper It should work

File: build.sbt [hunk 3]
```diff
@@ -21,11 +21,66 @@ lazy val sonataCredentials = for {
 
 credentials in ThisBuild ++= sonataCredentials.toSeq
 
+lazy val scalazDependencies = Seq(
+  "org.scalaz" %% "scalaz-core" % "7.2.25",
```
Comment (wi101) [inline build.sbt]: @quiquedeveloper We don't need scalaz-core

File: build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (wi101) [reply build.sbt]: I will take a look

File: build.sbt (hunk 3 above)
Comment (wi101) [reply build.sbt]: scalazDependencies will be used only in core, you can move it there

File: build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (wi101) [reply build.sbt]: you're right, we should add resolvers to every project

File: build.sbt [hunk 4]
```diff
@@ -21,11 +21,66 @@ lazy val sonataCredentials = for {
...
+  .settings(
+    resolvers += Resolver.sonatypeRepo("snapshots"),
+    skip in publish := true,
+    libraryDependencies ++=
+      Seq(
+        "org.scala-lang" % "scala-reflect"  % scalaVersion.value,
+        "org.scala-lang" % "scala-compiler" % scalaVersion.value % Provided,
+        "io.monix"       %% "monix"         % "3.0.0-RC1",
+        "org.typelevel"  %% "cats-effect"   % "1.0.0-RC2"
+      ) ++ scalazDependencies
```
[the remark this answers is not in the document]
Comment (quiquedeveloper) [reply build.sbt]: ok i removed it already

File: core/src/main/scala/scalaz/kleisliio/KleisliIO.scala (hunk 2 above)
Comment (quiquedeveloper) [reply core/src/main/scala/scalaz/kleisliio/KleisliIO.scala]: in the initial John's PR, KleisliIO was extending `Function1` but after some feedbacks he removed it. I guess he forgot to update the docu. we can update docu later

File: build.sbt (hunk 3 above)
Comment (quiquedeveloper) [reply build.sbt]: cool i removed scalazDependencies

**Pull request merged**